src/Compilers/Core/MSBuildTask/MvidReader.cs

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+
+namespace Microsoft.CodeAnalysis.BuildTasks
+{
+    public static class MvidReader
+    {
+        public static Guid ReadAssemblyMvidOrEmpty(Stream stream)
+        {
+            return ReadAssemblyMvidOrEmpty(new BinaryReader(stream));
+        }
+
+        private static Guid ReadAssemblyMvidOrEmpty(BinaryReader reader)
+        {
+            Guid empty = Guid.Empty;
+
+            // DOS Header (64), DOS Stub (64), PE Signature (4), COFF Header (20), PE Header (224), 1x Section Header (40)
+            if (reader.BaseStream.Length < 64 + 64 + 4 + 20 + 224 + 40)
+            {
+                return empty;
+            }
+
+            // DOS Header: Magic number (2)
+            if (reader.ReadUInt16() != 0x5a4d) // "MZ"
+            {
+                return empty;
+            }
+
+            // DOS Header: Address of PE Signature (at 0x3C)
+            MoveTo(0x3C, reader);
+            uint lfanew = reader.ReadUInt32();
+
+            MoveTo(lfanew, reader); // jump over the MS DOS Stub to the PE Signature
+
+            // PE Signature ('P' 'E' null null)
+            if (reader.ReadUInt32() != 0x00004550)
```

Comment:
>if (reader.ReadUInt32() != 0x00004550) [](start = 12, length = 38)

This looks  like an attempt to read at an unknown location, which can throw if we are near the end of the stream.